utils/active-hotspot-oracle/src/main.rs

```diff
@@ -0,0 +1,489 @@
+struct Args {
+  /// Bucket name for the store. Required
+  #[clap(long)]
+  pub aws_bucket: Option<String>,
+  #[clap(long)]
+  pub aws_table: Option<String>,
+  /// Optional api endpoint for the bucket. Default none
+  #[clap(long)]
+  pub aws_endpoint: Option<String>,
+  /// Optional region for the endpoint
+  #[clap(long)]
+  pub aws_region: Option<String>,
+
+  /// Should only be used for local testing, set as environment variables in deployment
+  #[clap(long)]
+  pub access_key_id: Option<String>,
+  #[clap(long)]
+  pub secret_access_key: Option<String>,
+
+  /// Solana paper wallet that will be the holder of the hotspot
+  #[arg(short, long)]
+  keypair: Option<String>,
+  /// RPC url
+  #[arg(short, long)]
+  url: Option<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct ActiveCheckpoint {
+  set: HashSet<String>,
+}
+
+fn load_arg_from_cli_or_env(cli_opt: Option<String>, env_name: &str) -> String {
+  let arg = if let Some(cli_arg) = cli_opt {
+    cli_arg
+  } else {
+    env::var(env_name)
+      .context(format!(
+        "Must supply {} as env var or cli argument",
+        env_name
+      ))
+      .unwrap()
+  };
+  arg
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+  std::env::set_var("RUST_LOG", "s3=debug");
+  use clap::Parser;
+  let args = Args::parse();
+
+  let kp_path = load_arg_from_cli_or_env(args.keypair, "ANCHOR_WALLET");
+  let rpc_url = load_arg_from_cli_or_env(args.url, "SOLANA_URL");
+  let s3_bucket = load_arg_from_cli_or_env(args.aws_bucket, "AWS_BUCKET");
+  let s3_table = load_arg_from_cli_or_env(args.aws_table, "AWS_TABLE");
+  let s3_region = load_arg_from_cli_or_env(args.aws_region, "AWS_REGION");
+  let access_key_id = load_arg_from_cli_or_env(args.access_key_id, "AWS_ACCESS_KEY_ID");
+  let secret_access_key = load_arg_from_cli_or_env(args.secret_access_key, "AWS_SECRET_ACCESS_KEY");
+
+  // Create anchor program client
+  let kp = Rc::new(read_keypair_file(kp_path).unwrap());
+  let anchor_client = Client::new_with_options(
+    Cluster::Custom(
+      rpc_url.clone(),
+      rpc_url
+        .clone()
+        .replace("https", "wss")
+        .replace("http", "ws"),
+    ),
+    kp.clone(),
+    CommitmentConfig::confirmed(),
+  );
+  let helium_entity_program = anchor_client.program(helium_entity_manager::id());
+  let tpu_client = TpuClient::new(
+    helium_entity_program.rpc().into(),
+    &helium_entity_program
+      .rpc()
+      .url()
+      .replace("https", "wss")
+      .replace("http", "ws"),
+    TpuClientConfig::default(),
+  )
+  .unwrap();
+
+  // set up delta-rs connection
+  let table_path = format!("s3://{}/{}", s3_bucket, s3_table);
+  let mut s3_config = HashMap::from([("aws_default_region".to_string(), s3_region.clone())]);
+  s3_config.insert("bucket_name".to_string(), s3_bucket);
+  s3_config.insert("aws_secret_access_key".to_string(), secret_access_key);
+  s3_config.insert("allow_http".to_string(), "true".to_string());
```

Comment:
These configs should prob be different in production. Make it conditional on the secret/access keys being present? Look at how delta lake sink does this